google-api-client/src/main/java/com/google/api/client/googleapis/apache/v2/GoogleApacheHttpTransport.java

```diff
@@ -41,9 +43,31 @@
   /**
    * Returns a new instance of {@link ApacheHttpTransport} that uses
    * {@link GoogleUtils#getCertificateTrustStore()} for the trusted certificates.
+   * If `GOOGLE_API_USE_CLIENT_CERTIFICATE` environment variable is set to "true",
+   * and the default client certificate from {@link Utils#loadDefaultCertificate()}
+   * is not null, then the transport uses the default client certificate and
+   * is mutual TLS.   
    */
-  public static ApacheHttpTransport newTrustedTransport() throws GeneralSecurityException,
-      IOException {
+  public static ApacheHttpTransport newTrustedTransport()
+      throws GeneralSecurityException, IOException {
+    return newTrustedTransport(null);
+  }
+
+  /**
+   * {@link Beta} <br>
+   * Returns a new instance of {@link ApacheHttpTransport} that uses
+   * {@link GoogleUtils#getCertificateTrustStore()} for the trusted certificates.
+   * If `GOOGLE_API_USE_CLIENT_CERTIFICATE` environment variable is set to "true",
+   * the function looks for user provided client certificate first from 
+   * clientCertificateSource InputStream, if not exists, then the default from
+   * {@link Utils#loadDefaultCertificate()}. If client certificate exists,
+   * the transport uses it and is mutual TLS.
+   * 
+   * @param clientCertificateSource InputStream for mutual TLS client certificate and private key   
+   */
+  @Beta
+  public static ApacheHttpTransport newTrustedTransport(InputStream clientCertificateSource)
```

Comment:
I don't think we need to make this public. If we need to allow the user to customize the mtls credentials, we should directly pass through the `KeyStore` and password without requiring an `InputStream`. We are already providing helpers to load a `KeyStore` from `InputStream`.

You can use the default visibility if you want to dependency inject for testing.

Reply:
now using keystore/password